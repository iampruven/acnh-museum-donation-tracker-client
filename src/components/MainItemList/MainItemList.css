ul li {
    list-style-type: none;  
}
i{
    cursor: pointer;
}
li{
    display:flex;
    align-items: center;
}
.center{
    display:flex;
    justify-content: center;
}

.backdrop{
    background-color: white;
    border-radius: 20px;
    border:4px gold solid;
    margin-left:20px;
    margin-right:20px;
    position: absolute;
    left: 11%;
    width: 71%;

}
.title-links a{
    font-size: 18px;
    font-weight: 900;
    padding: 10px;
}
.title-links a:active, .title-links a:hover, .title-links a:focus {
color:#517664;
}
@media only screen and (min-width: 900px) {
    .backdrop{
        background-color: white;
        border-radius: 20px;
        border:4px gold solid;
        max-width: 25%;
        left: 36%;
        position: absolute;

    
    }

  }