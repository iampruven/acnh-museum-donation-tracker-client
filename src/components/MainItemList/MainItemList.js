import React from "react";
import config from "../../config";
import TokenService from '../../services/token-service';
export default class MainPage extends React.Component {
  state = {
    option: "all",
    items: [],
  };

  componentDidMount() {
    fetch(`${config.API_ENDPOINT}/api/items`, {
      headers: {
        'authorization': `basic ${TokenService.getAuthToken()}`,
      },
    })
      .then((res) => {
        if (!res.ok) {
          throw new Error(res.statusText);
        }
        return res.json();
      })
      .then((data) => {
        console.log(data);
        this.setState({ items: data });
      })
      .catch((err) => err.message);
  }

  handleDeleteRequest(e,id){
    e.preventDefault();
    fetch(`${config.API_ENDPOINT}/api/items/${id}`,{
      method: 'DELETE',
      headers: {
        'Content-Type': 'application/json',
        'authorization': `basic ${TokenService.getAuthToken()}`,
      },
    })
      .then(res=>{
        if(!res.ok)
          return res.json().then(e=>Promise.reject(e))
      })
      .then(()=>{
        this.handleDeleteItem(id)
      })
      .catch(e=>console.log({e}))
     
  }
  handleDisplayType = (type) => () => {
    this.setState({
      option: type,
    });
  };

  handleDeleteItem = (id) => {
    this.setState({
      items: this.state.items.filter((items) => items.id !== id),
    });
  };


  render() {
    console.log(this.state.items)
    const data =
      this.state.option === "all"
        ? this.state.items.map((x) => (
            <li>
              <img alt={x.name} src={x.img} /> {x.name} {x.date}{" "}
              <span onClick={(e) => this.handleDeleteRequest(e,x.id)}>
                <i class="fa fa-trash" aria-hidden="true"></i>
              </span>
            </li>
          ))
        : this.state.items
            .filter((m) => m.type === this.state.option)
            .map((x) => (
              <li>
                <img alt={x.name} src={x.img} /> {x.name} {x.date}{" "}
                <span onClick={(e) => this.handleDeleteRequest(e,x.id)}>
                  <i class="fa fa-trash" aria-hidden="true"></i>
                </span>
              </li>
            ));
    return (
      <section>
        <div>
          <a onClick={this.handleDisplayType("Bugs")} href="#">
            Bugs
          </a>
          <a onClick={this.handleDisplayType("Fossils")} href="#">
            Fossils
          </a>
          <a onClick={this.handleDisplayType("Fish")} href="#">
            Fish
          </a>
          <a onClick={this.handleDisplayType("all")} href="#">
            All
          </a>
        </div>
        <div>{data}</div>
      </section>
    );
  }
}
