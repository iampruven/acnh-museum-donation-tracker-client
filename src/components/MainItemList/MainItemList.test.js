import React from 'react';
import ReactDOM from 'react-dom';
import MainItemList from './MainItemList';
import renderer from 'react-test-renderer';

//smoke test
describe('renders without crashing',()=>{
    it('it renders without crashing',()=>{
        const div = document.createElement('div');

        ReactDOM.render(<MainItemList />,div);
        ReactDOM.unmountComponentAtNode(div);
    })


//snapshot test
    it('renders the UI as expected', ()=>{
        const tree = renderer
            .create(<MainItemList />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })



})