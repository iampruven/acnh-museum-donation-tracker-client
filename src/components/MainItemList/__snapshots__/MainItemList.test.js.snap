// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing renders the UI as expected 1`] = `
<section
  className="backdrop"
>
  <div
    className="center title-links"
  >
    <a
      href="#"
      onClick={[Function]}
    >
      Bugs
    </a>
    <a
      href="#"
      onClick={[Function]}
    >
      Fossils
    </a>
    <a
      href="#"
      onClick={[Function]}
    >
      Fish
    </a>
    <a
      href="#"
      onClick={[Function]}
    >
      All
    </a>
  </div>
  <div
    className="center"
  >
    <ul />
  </div>
</section>
`;
